### PR TITLE
feat(remix-react): clean up which button clicked submit implementation

### PR DESCRIPTION
**Pull request by mcansh**
Signed-off-by: Logan McAnsh <logan@mcan.sh>

### CODE REVIEW

File: packages/remix-react/components.tsx [hunk 1]
```diff
@@ -1105,6 +1069,7 @@ export function useSubmitImpl(key?: string): SubmitFunction {
           options.encType || target.getAttribute("enctype") || defaultEncType;
 
         formData = new FormData(target);
+        console.log(Object.fromEntries(formData));
```
Comment (mjackson) [inline packages/remix-react/components.tsx:1072]: Remove

File: packages/remix-react/components.tsx [hunk 2]
```diff
@@ -970,11 +926,13 @@ let FormImpl = React.forwardRef<HTMLFormElement, FormImplProps>(
...
 
-                submit(clickedButtonRef.current || event.currentTarget, {
-                  method,
-                  replace,
-                });
-                clickedButtonRef.current = null;
+                let submitter = (event as unknown as HTMLSubmitEvent)
+                  .nativeEvent.submitter as
+                  | HTMLButtonElement
+                  | HTMLInputElement
```
Comment (mjackson) [inline packages/remix-react/components.tsx:932]: Let's create a separate type for this and use it here.

```ts
type HTMLFormSubmitter = HTMLButtonElement | HTMLInputElement;
```

**Review by mjackson — CHANGES_REQUESTED** [review]
Let's also remove the unused `useComposedRefs` function in this file.

File: packages/remix-react/components.tsx (hunk 1 above)
Comment (mcansh) [reply packages/remix-react/components.tsx:1072]: weird, i thought i did!

**Review by mjackson — APPROVED** [review]

**Pull request merged**